desk/sur/forums.hoon

```diff
@@ -138,99 +118,125 @@
-        %+  levy
-          tags
-        |=  a=term 
-        ?~((find ~[a] allowed-tags.metadata.rock) %| %&)
+      =/  tagset=(set term)  (silt tags.act)
+      ?.  (are-tags-valid tagset allowed-tags.metadata.rock)
+        ~|('%forums: attempting to add one or more disallowed tags' !!)
       ::  Insert a new entry into threads
-      =.  database.rock  
+      =.  database.rock
+        =/  new-thread=thread
+        :~  next-id.metadata.rock  [%s ~]
+            title.act  [%s tagset]
+        ==
         %+  ~(insert-rows db database.rock)
-          %forums^thread-table 
-        ~[[count.metadata.rock now.bowl src.bowl title.act tags.act]]
-      ::  Insert a new entry into posts, update board counter
+          %forums^thread-table
+        ~[new-thread]
+      ::  Insert a new entry into posts, update board next ID
       =.  database.rock
         =/  new-post=post
-        :~  count.metadata.rock  count.metadata.rock  ~
-            now.bowl  src.bowl  content.act
-            [%s ~]  [%b ~]  [%m ~]  [%s ~]
+        :~  next-id.metadata.rock  ~
+            [%s ~]  [%m ~]
+            [%b (put:om-hist *edits now.bowl [src.bowl content.act])]
         ==
         %+  ~(insert-rows db database.rock)
-          %forums^post-table  
+          %forums^post-table
         ~[new-post]
-      =.  count.metadata.rock  +(count.metadata.rock)
+      =.  next-id.metadata.rock  +(next-id.metadata.rock)
       rock
     ::
         %new-reply
       ::  Make sure that the thread exists
-      ?<  .=(~ (get-thread database.rock thread-table thread-id.act))
+      ?<  .=(~ (get-thread database.rock thread-table parent-id.act))
       =/  new-post=post
-      :~  count.metadata.rock  thread-id.act  parent-id.act
-          now.bowl  src.bowl  content.act
-          [%s ~]  [%b ~]  [%m ~]  [%s ~]
+      :~  next-id.metadata.rock  parent-id.act
+          [%s ~]  [%m ~]
+          [%b (put:om-hist *edits now.bowl [src.bowl content.act])]
       ==
-      ::  Check for repeat post
+      ::  Check for repeat post (only one response per author per thread)
       ~|  '%forums: cannot post twice in same thread'
-      ?>    
+      ?>
       .=  ~
       %+  turn
         =<  -
         %+  ~(q db database.rock)
           %forums
-        :+  %select 
+        :+  %select
           post-table
-        :+  %and 
-          [%s %thread-id [%& %eq thread-id.act]]
-        [%s %author [%& %eq src.bowl]]
-      |=  =row 
-      !<(post [-:!>(*post) row])
+        :+  %and
+          [%s %parent-id [%& %eq parent-id.act]]
+        :^  %s  %history  %|
+        |=  =value
+        ?>  ?=([%b *] value)
+        =+  ;;(edits p.value)
+        =/  [[@da author=@p @t] edits]  (pop:om-hist -)
+        .=(author src.bowl)
+      |=(=row !<(post [-:!>(*post) row]))
       ::  Insert a new entry into posts table
       =.  database.rock
         %+  ~(insert-rows db database.rock)
         %forums^post-table  ~[new-post]
```

Comment:
```suggestion
        %+  ~(insert-rows db database.rock)
          %forums^post-table
        ~[new-post]
```

this was my code, but putting a note here because I realized that the formatting is wrong